userspace/libscap/scap_procs.c

```diff
@@ -403,6 +403,72 @@ static int32_t scap_proc_fill_flimit(scap_t *handle, uint64_t tid, struct scap_t
 }
 #endif
 
+int32_t scap_proc_fill_overlayfs_root(scap_t *handle, struct scap_threadinfo *tinfo, const char *procdirname)
+{
+	FILE *mounts_file;
+	char mounts_path[SCAP_MAX_PATH_SIZE];
+	char line[SCAP_MAX_ARGS_SIZE];
+
+	tinfo->overlayfs_root[0] = '\0';
+
+	snprintf(mounts_path, sizeof(mounts_path), "%smounts", procdirname);
+	mounts_file = fopen(mounts_path, "r");
+	if(!mounts_file)
+	{
+		return SCAP_FAILURE;
+	}
+
+	if(!fgets(line, sizeof(line), mounts_file))
+	{
+		fclose(mounts_file);
+		return SCAP_FAILURE;
+	}
+
+	fclose(mounts_file);
+
+	if(strstr(line, "overlay / overlay"))
+	{
+		char *start, *end;
+
+		//
+		//  Get the lower dir
+		//
+		start = strstr(line, "lowerdir=") + sizeof("lowerdir=") - 1;
+		if(!start)
```

Comment:
I don't think this check will work, you are using pointer arithmetic in a way that will cause `start` to never be `NULL` even if `strstr` fails.

I suggest first calling `start = strstr(line, "lowerdir=")`, check it for null and then adjust its position.